### ISSUE TITLE
Extension activationEvent onLanguage:plaintext always fires, even if no plaintext files are open or a non-plaintext file is opened

### ISSUE THREAD

**Issue by valentjn** [microsoft/vscode#80268]
Issue Type: <b>Bug</b>

## Version info:
```
Version: 1.37.1
Commit: f06011ac164ae4dc8e753a3fe7f9549844d15e35
Date: 2019-08-15T16:17:25.463Z
Electron: 4.2.7
Chrome: 69.0.3497.128
Node.js: 10.11.0
V8: 6.9.427.31-electron.0
OS: Linux x64 4.15.0-55-generic
```

## Steps to reproduce:
1. Download Minimal Hello World Sample extension from https://github.com/microsoft/vscode-extension-samples/tree/master/helloworld-minimal-sample to `~/.vscode/extensions/helloworld-minimal-sample`.
2. Replace the `onCommand:extension.helloWorld` activation event in `~/.vscode/extensions/helloworld-minimal-sample/package.json` with `onLanguage:plaintext`.
3. Start VS Code via `code`.
4. Select View -> Output -> Log (Extension Host).

## Actual behavior:
The log contains the line
```
[2019-09-03 16:44:58.410] [exthost] [info] ExtensionService#_doActivateExtension vscode-samples.helloworld-minimal-sample {"startup":false,"activationEvent":"onLanguage:plaintext"}
```
i.e., the Minimal Hello World Sample extension is activated.

## Expected behavior:
The extension is not activated, since no plaintext files are open.

## Additional comments:
* Even if the Welcome screen is closed, a `*.cpp` file is opened, VS Code is reloaded again, VS Code still activates the extension. If all tabs are closed and VS Code is reloaded, the extension doesn't get activated. However, when then opening a `*.cpp` file, the extension will be activated again.
* Probably related to #28732 and #61825.

VS Code version: Code 1.37.1 (f06011ac164ae4dc8e753a3fe7f9549844d15e35, 2019-08-15T16:17:25.463Z)
OS version: Linux x64 4.15.0-55-generic

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-4300U CPU @ 1.90GHz (4 x 2771)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: disabled_software<br>skia_deferred_display_list: disabled_off<br>skia_renderer: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: unavailable_off<br>viz_display_compositor: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 1, 1|
|Memory (System)|11.43GB (2.32GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (1)</summary>

Extension|Author (truncated)|Version
---|---|---
helloworld-minimal-sample|vsc|0.0.1

</details>

**Comment by alexandrudima** [comment]
I am not sure if this is intentional or not. I see the following stack triggering the activation event:
```
    at ExtensionService.activateByEvent (abstractExtensionService.js:127)
    at WorkbenchModeServiceImpl.onDidCreateMode (workbenchModeService.js:124)
    at Emitter.fire (event.js:455)
    at WorkbenchModeServiceImpl._getOrCreateMode (modeServiceImpl.js:128)
    at WorkbenchModeServiceImpl._createModeAndGetLanguageIdentifier (modeServiceImpl.js:106)
    at LanguageSelection [as _selector] (modeServiceImpl.js:100)
    at new LanguageSelection (modeServiceImpl.js:14)
    at WorkbenchModeServiceImpl.createByFilepathOrFirstLine (modeServiceImpl.js:98)
    at content.then.content (walkThroughContentProvider.js:40)
```

**Comment by valentjn** [comment]
So this is probably caused by this:
https://github.com/microsoft/vscode/blob/0f1bd7721b793ed8b7805ba621061ce5a1f5c263/src/vs/editor/common/services/modeServiceImpl.ts#L148-L153
Even if it is intentional, is there a way to deactivate an extension that has been activated by `onLanguage:plaintext` if the document turns out not be plaintext?

**Comment by chrmarti** [comment]
@alexandrudima Not intentional. Should I add a language to avoid triggering well-known activation events or how can I avoid that?

**Comment by alexandrudima** [comment]
@chrmarti I think it should be possible to avoid instantiating a `TextModel` in order to render a custom workbench editor... I think @bpasero would know...

**Comment by bpasero** [comment]
Yes in theory, you are already creating you own custom editor I suppose, so you could probably use the code editor as widget and fill in the editor value. How are you doing this today?

**Comment by chrmarti** [comment]
Not sure, this might be ancient code. Do I need an input and input provider: https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/contrib/welcome/page/browser/welcomePage.ts#L274 ?

**Comment by bpasero** [comment]
@chrmarti if you are going through the workbench model factory ([here](https://github.com/microsoft/vscode/blob/afffb1e7a77664632e1e6a5079b7e419f015ab1f/src/vs/workbench/contrib/welcome/walkThrough/browser/walkThroughInput.ts#L98)), you will always end up with a workbench text model (which probably triggers the activation). 

I currently do not see an alternative unless you really want to just put the editor control into the dom and provide input and colorization yourself (e.g. from https://github.com/microsoft/monaco-editor-samples/blob/master/browser-script-editor/index.html).

**Comment by bpasero** [comment]
Btw I do not understand why the `walkThroughContentProvider` is up and running on startup when it should only ever be up and running when I open the "Interactive Editor Playground".

**Comment by chrmarti** [comment]
I'm reusing that code for the Welcome page. I guess for the embedded editors in the Playground it makes sense to participate in the workbench text models to get all the commands. For the Welcome page and the Playground's workbench editor this might not be necessary.
